File: export.py
from bs4 import BeautifulSoup
import jsmin

def extract_and_minify_js(html_file):
    with open(html_file, 'r', encoding='utf-8') as file:
        html_content = file.read()

    soup = BeautifulSoup(html_content, 'html.parser')

    js_code = ''
    for script_tag in soup.find_all('script'):
        js_code += script_tag.get_text()


    minified_js = jsmin.jsmin(js_code)

    minified_js = minified_js.replace('\n', '').replace('\r', '')
    minified_js = minified_js.replace('"', "'")

    minified_js +="virtualKeyPress();"

    minified_js_string = f'"{minified_js}"'

    print(minified_js_string)

extract_and_minify_js('index.html')

File: export.js
const fs = require('fs');
const { minify } = require('terser');
const { JSDOM } = require('jsdom');


async function extractAndMinifyJs(htmlFile, outputFile) {
    const htmlContent = fs.readFileSync(htmlFile, 'utf-8');

    const dom = new JSDOM(htmlContent);
    const scripts = dom.window.document.querySelectorAll('script');

    let jsCode = '';
    scripts.forEach(script => {
        jsCode += script.textContent;
    });

    const { code: minifiedJs } = await minify(jsCode, {
        output: { quote_style: 1 },
        mangle: {
            toplevel: true
        },
    });

    const minifiedJsString = `"${minifiedJs.replace(/\n|\r/g, '').replace(/"/g, "'")}"`;
    console.log(minifiedJsString);

    const updatedHtmlContent = htmlContent.replace(/<script>[\s\S]*?<\/script>/g, '');
    const virtualHtmlContent = updatedHtmlContent.replace(/onClick="plugboardPress\(\)"/g, `onClick="virtualKeyPress()"`);
    const finalHtmlContent = virtualHtmlContent.replace(/onClick="virtualKeyPress\(\)"/g, `onClick=${minifiedJsString}`);

    fs.writeFileSync(outputFile, finalHtmlContent, 'utf-8');
}

extractAndMinifyJs('index.html', 'output.html');